### ISSUE TITLE
discrepency in you AlexNet model

### ISSUE THREAD

**Issue by Coderx7** [kumar-shridhar/PyTorch-BayesianCNN#25]
Hello. 
I noticed you didnt include the kl term for the CNN layers in your loss. was this intentional since the previous AlexNet model, still uses that term!

**Comment by kumar-shridhar** [comment]
Hi @Coderx7 ,

Can you please be a bit more specific as this term in Conv layer: 
![image](https://user-images.githubusercontent.com/20888097/64875079-4ebcca80-d61a-11e9-9618-74e3144b336e.png)
includes the KL in the overall loss.

**Comment by Coderx7** [comment]
Hi, in [Image Recognition/utils/BayesianModels/BayesianAlexNet.py](https://github.com/kumar-shridhar/PyTorch-BayesianCNN/blob/master/Image%20Recognition/utils/BayesianModels/BayesianAlexNet.py#L48), if you look closely you can see that, you are actually only taking into account the kl regarding the fc layers (i.e. the classifier). 
```python
def probforward(self, x):
    kl = 0
    for layer in self.layers:
        if hasattr(layer, 'convprobforward') and callable(layer.convprobforward):
            x, _kl, = layer.convprobforward(x)
        else:
            x = layer.forward(x)
    x = x.view(x.size(0), -1)
    x, _kl = self.classifier.fcprobforward(x)
    kl += _kl
    logits = x
    return logits, kl
```
the line 
`        x, _kl = self.classifier.fcprobforward(x)`
overwrites the previous _kl from the convlayer . this however is not the case in `BayesianAlexNet_.py` and in that file, the kl is accounted for, for both types of layers.

**Comment by JurijsNazarovs** [comment]
Hello @Coderx7, did you try to fix it with kl += _kl as it was before? If yes, is result from paper still reproducible?

**Comment by Coderx7** [comment]
@JurijsNazarovs I didnt run the code, I was just eyeing through code and spotted that. 
by the way please accept my apologies for replying this late I just forgot about this!

**Comment by suredie** [comment]
> Hello @Coderx7, did you try to fix it with kl += _kl as it was before? If yes, is result from the paper still reproducible?

No, by doing so, it seems the model cannot converge.

**Comment by kumar-shridhar** [comment]
Hi, 
Sorry I was busy with some other commitments. I have tried to fix this issue. Please check it out and let me know your thoughts.
